### PR TITLE
libtiff: 4.0.10 -> 4.1.0

### LINKED ISSUE #57158: Vulnerability roundup 63: libtiff-4.0.10: 2 advisories
[search](https://search.nix.gsc.io/?q=libtiff&i=fosho&repos=nixos-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=libtiff+in%3Apath&type=Code)

* [x] [CVE-2019-6128](https://nvd.nist.gov/vuln/detail/CVE-2019-6128) (nixos-19.03)
* [x] [CVE-2019-7663](https://nvd.nist.gov/vuln/detail/CVE-2019-7663) (nixos-19.03)

Scanned versions: nixos-19.03: 5847485e3ec. May contain false positives.

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
https://nvd.nist.gov/vuln/detail/CVE-2019-7663

vulnerable in unstable, 19.03 and 19.09

Fixes #57158

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

No maintainer :(

### CODE REVIEW

File: pkgs/development/libraries/libtiff/default.nix [hunk 1]
```diff
@@ -23,6 +23,11 @@ stdenv.mkDerivation rec {
       name = "CVE-2019-6128.patch";
       sha256 = "03yvsfq6dxjd3v8ypfwz6cpz2iymqwcbawqqlmkh40dayi7fgizr";
     })
+    (fetchurl {
+      url = "https://gitlab.com/libtiff/libtiff/commit/802d3cbf3043be5dce5317e140ccb1c17a6a2d39.patch";
+      name = "CVE-2019-7663";
```
Comment (mmahut) [inline pkgs/development/libraries/libtiff/default.nix]:
```suggestion
      name = "CVE-2019-7663.patch";
```

**Comment by mmahut** [comment]
This should probably go into staging for mass rebuild.

**Comment by FRidh** [comment]
@JohnAZoidberg you may want to enable "Allow edits from maintainers" as that makes it easier for others to pick up your changes, modify them when needed, and submit them. Of course you do not have to if you do not want to!
https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by dtzWill** [comment]
Perhaps update to 4.1.0 as well (instead?)? Didn't check if contains fix but seems likely :).

**Comment by JohnAZoidberg** [comment]
"Allow edits from maintainers" is enabled, like always by default. Did you try pushing something to my branch and it didn't work?
Sorry not coming back to this PR.

Yes, 4.1.0 includes all patches. Should we backport the new version?
Skimming through the [changelog](https://gitlab.com/libtiff/libtiff/blob/master/ChangeLog) I don't see any immediately obvious incompatibilities but I'm not sure about some changes.
So I'd suggest to backport just the patching commit.

**Review by vcunat — APPROVED** [review]

**Pull request merged**